factory.py

```diff
@@ -23,17 +31,28 @@ def get_factories(task: Union[str, FactoryTypes, Tasks]) -> List["Factory"]:
         List[Factory]: List of factory classes.
     """
 
-    task: FactoryTypes = FactoryTypes(task)
-    if task == FactoryTypes.TEXT_GENERATION:
-        # text-generation
-        from text_generation.main import Factory as TextGenerationFactory
-
-        cls = [TextGenerationFactory]
-    elif task == FactoryTypes.VISION:
-        # vision
-        from vision.factory import CompressionFactory as VisionFactory
+    if not job_service or job_service == JobServices.KOMPRESS:
```

Comment:
this was done because kompress yamls don't have a default argument called job_service (or atleast they didn't while this change was being made). So, for this to change, I think this has to pe another PR with the updated YAMLs as well, especially for text-generation

Reply:
text-generation has yamls with a job service key. And this change will ensure we catch the yamls without a job service key.

Reply:
Vision yamls still don't use a JOB_SERVICE. Anyway, I will update this.

Reply:
This is done now